### PR TITLE
Add Search Functionality and Limits

### PR DESCRIPTION
**Pull request by zun1uwu**

### CODE REVIEW

File: src/routes/Library.svelte [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+			>
+		</div>
+
+		<!-- Awaits for games to be resolved -->
+		<!-- After that, loop over every object in that array -->
+		<!-- And add those results to a div -->
+		<!-- svelte-ignore empty-block -->
+		{#await games then data}
+			{#each data as game}
+				{#if !query}
```
Comment (infinity-plus) [inline src/routes/Library.svelte:75]: This thing can be simplified and the code can be reduced to half if we implement a function that returns a new array of filtered results matching the query, or returning the whole array if query is empty.

Use this demo function:
```typescript
function getFilteredGames(games: array[], gameToSearch?: string){
    if(typeof gameToSearch !== 'undefined'){
        return games.filter(game => {
            return game.title.toLowerCase().includes(gameToSearch.toLowerCase());
        });
    } else {
        return games;
    }
}
```

The above function is not tested by me. Please do necessary checks.

**Review by infinity-plus — CHANGES_REQUESTED** [review]
Refer to this comment and please make changes if necessary:

https://github.com/ProjectBlackPearl/pbpl/pull/21/files#r1070355763

**Comment by Brisolo32** [comment]
I'll wait until the action is done, but overall LGTM

**Review by infinity-plus — APPROVED** [review]

**Review by Brisolo32 — APPROVED** [review]
LGTM

**Pull request merged**